libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/maneuver/ManeuverViewHelper.kt

```diff
@@ -0,0 +1,166 @@
+            put(Pair(null, STEP_MANEUVER_MODIFIER_SLIGHT_LEFT),
+                    ManeuverViewUpdate { canvas, primaryColor, secondaryColor, size, roundaboutAngle ->
+                        ManeuversStyleKit.drawArrowSlightRight(canvas, primaryColor, size)
+                    })
+            put(Pair(null, STEP_MANEUVER_MODIFIER_LEFT),
+                    ManeuverViewUpdate { canvas, primaryColor, secondaryColor, size, roundaboutAngle ->
+                        ManeuversStyleKit.drawArrowRight(canvas, primaryColor, size)
+                    })
+            put(Pair(null, STEP_MANEUVER_MODIFIER_SHARP_LEFT),
+                    ManeuverViewUpdate { canvas, primaryColor, secondaryColor, size, roundaboutAngle ->
+                        ManeuversStyleKit.drawArrowSharpRight(canvas, primaryColor, size)
+                    })
+            put(Pair(null, STEP_MANEUVER_MODIFIER_UTURN),
+                    ManeuverViewUpdate { canvas, primaryColor, secondaryColor, size, roundaboutAngle ->
+                        ManeuversStyleKit.drawArrow180Right(canvas, primaryColor, size)
+                    })
+            put(Pair(null, STEP_MANEUVER_MODIFIER_STRAIGHT),
+                    ManeuverViewUpdate { canvas, primaryColor, secondaryColor, size, roundaboutAngle ->
+                        ManeuversStyleKit.drawArrowStraight(canvas, primaryColor, size)
+                    })
+            put(Pair(null, null),
+                    ManeuverViewUpdate { canvas, primaryColor, secondaryColor, size, roundaboutAngle ->
+                        ManeuversStyleKit.drawArrowStraight(canvas, primaryColor, size)
+                    })
+        }
+    }
+
+    @JvmField
+    val SHOULD_FLIP_MODIFIERS: Set<String> = object : HashSet<String>() {
+        init {
+            add(STEP_MANEUVER_MODIFIER_SLIGHT_LEFT)
+            add(STEP_MANEUVER_MODIFIER_LEFT)
+            add(STEP_MANEUVER_MODIFIER_SHARP_LEFT)
+            add(STEP_MANEUVER_MODIFIER_UTURN)
+        }
+    }
+
+    @JvmField
+    val ROUNDABOUT_MANEUVER_TYPES: Set<String> = object : HashSet<String>() {
+        init {
+            add(STEP_MANEUVER_TYPE_ROTARY)
+            add(STEP_MANEUVER_TYPE_ROUNDABOUT)
+            add(STEP_MANEUVER_TYPE_ROUNDABOUT_TURN)
+            add(STEP_MANEUVER_TYPE_EXIT_ROUNDABOUT)
+            add(STEP_MANEUVER_TYPE_EXIT_ROTARY)
+        }
+    }
+
+    @JvmField
+    val MANEUVER_TYPES_WITH_NULL_MODIFIERS: Set<String> = object : HashSet<String>() {
+        init {
+            add(STEP_MANEUVER_TYPE_OFF_RAMP)
+            add(STEP_MANEUVER_TYPE_FORK)
+            add(STEP_MANEUVER_TYPE_ROUNDABOUT)
+            add(STEP_MANEUVER_TYPE_ROUNDABOUT_TURN)
+            add(STEP_MANEUVER_TYPE_EXIT_ROUNDABOUT)
+            add(STEP_MANEUVER_TYPE_ROTARY)
+            add(STEP_MANEUVER_TYPE_EXIT_ROTARY)
+        }
+    }
+
+    @JvmStatic
+    fun isManeuverIconNeedFlip(maneuverType: String?, maneuverModifier: String?, drivingSide: String?): Boolean {
```

Comment:
This method is from the original java version:
https://github.com/mapbox/mapbox-navigation-android/commit/2d13a1edbb7e3ef96595d60eb7945ce3381eac04#diff-3dd5341892b02d8c39fe0b1b5b02de9dR238-R246
I don't quite sure the logic here, but checking with @Guardiola31337 that since it works I only convert it to Kotlin here and doesn't change any logic.